sentry/src/main/java/io/sentry/protocol/Request.java

```diff
@@ -159,9 +163,38 @@ public void setOthers(Map<String, String> other) {
     this.other = other;
   }
 
+  /**
+   * the Request's unknown fields
+   *
+   * @return the unknown map
+   */
+  @TestOnly
+  @Nullable
+  Map<String, Object> getUnknown() {
+    return unknown;
+  }
+
   @ApiStatus.Internal
   @Override
   public void acceptUnknownProperties(Map<String, Object> unknown) {
     this.unknown = unknown;
   }
+
+  /**
+   * Clones an User aka deep copy
+   *
+   * @return the cloned User
+   * @throws CloneNotSupportedException if the User is not cloneable
+   */
+  @Override
+  public @NotNull Request clone() throws CloneNotSupportedException {
+    final Request clone = (Request) super.clone();
+
+    clone.headers = CollectionUtils.shallowCopy(headers);
```

Comment:
I think `clone.data` could be a List or Map and it might require a shallow copy, I guess we've done this in another clone methods.

Reply:
you mean to check if with `instanceof`? Since `data` can be anything for now it's just a reference copy.

Reply:
yes, but if it's a list/map and items are added/removed, it would mutate both, right, that's what we could avoid.
we did something similar here https://github.com/getsentry/sentry-dart/blob/main/dart/lib/src/protocol/request.dart#L25-L38